iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/resource/tsfile/PipeTsFileResourceManager.java

```diff
@@ -176,19 +232,83 @@ public synchronized void decreaseFileReference(File hardlinkOrCopiedFile) throws
    * @param hardlinkOrCopiedFile the copied or hardlinked file
    * @return the reference count of the file
    */
-  public synchronized int getFileReferenceCount(File hardlinkOrCopiedFile) {
-    return hardlinkOrCopiedFileToReferenceMap.getOrDefault(hardlinkOrCopiedFile.getPath(), 0);
+  public int getFileReferenceCount(File hardlinkOrCopiedFile) {
+    lock.lock();
+    try {
+      final String filePath = hardlinkOrCopiedFile.getPath();
+      final PipeTsFileResource resource = hardlinkOrCopiedFileToPipeTsFileResourceMap.get(filePath);
+      return resource != null ? resource.getReferenceCount() : 0;
+    } finally {
+      lock.unlock();
+    }
+  }
+
+  public boolean tryCacheObjects(File hardlinkOrCopiedTsFile) throws IOException {
```

Comment:
It may be confused when using "tryCacheObjects" to detect whether the object is cached. Consider divide a new method from it?

Reply:
I think so